internal/mutex/mutex.go

```diff
@@ -0,0 +1,33 @@
+package mutex
+
+import "sync"
+
+// Mutex is a simple mutex that can be used to protect a value.
+//
+// The zero value is safe to use if the zero value of T is safe to use.
+//
+// Example:
+//
+//	var m mutex.Mutex[*string]
+//	s := m.Lock()
+//	defer m.Unlock()
+//	*s = "hello"
+type Mutex[T any] struct {
```

Comment:
Nice, this seems handy! I was going to ask why no use xsync, but this is more generic it seems.